Feature/cards

pierrefortier opened the pull request:
Created a new template for cards, for issue #3.

pierrefortier committed:
Initial card molecule commit

pierrefortier committed:
card styling updates

mel-miller commented:
@pierrefortier -- This looks good visually, but we'll need two structural changes before merging this in. We can discuss further at meeting. 

1. Separate the single card and card grid out into separate patterns. The "Card" can stay where it is, and the "Card Grid" should be under organisms.
2. Pull the sample data out into a yml file and use twig variables to pull them back into the markup. This part is necessary for us to use in CMS themes. You should also create the variants (various sizes, maybe different background color, etc...)  this way. The data structure will be a little more complex for the card grid, but you can look at [this file](https://github.com/it-osu-web/it-osu-pl/blob/develop/components/_patterns/02-molecules/social-media-links/social-media-links.yml) for an example of how it works where there are a series of items.

We will eventually build a basic grid pattern into the base category (I'm deciding on a few specifics now), which can get pulled into other components that will use a grid, but for now, how you've done it with flexbox in the component's css is good.

mel-miller commented:
Actually, the [social media links pattern](https://github.com/it-osu-web/it-osu-pl/blob/develop/components/_patterns/02-molecules/social-media-links/social-media-links.twig) is also a good example for including a single pattern into a series with a for loop.

pierrefortier committed:
Moved card-grid into an organism
Updated styling
Cleaned up code

pierrefortier commented:
I've updated the branch with a separate organism for the card grid. I also tweaked some styling and cleaned up the code a bit. The card variables are now in a separate twig file, and the different sizes are dealt with through variants.

pds-mel-bot committed:
Merge branch 'develop' into feature/cards

pds-mel-bot committed:
move card-grid styles to reside in card-grid component

mel-miller commented:
@pierrefortier — this is very close. I just wanted to discuss margins and applying them by context.

pierrefortier committed:
Removed margin from card molecule and added it to the grid organism

pierrefortier commented:
@mel-miller I just pushed an update that removed the margins from the card molecule and added it to the grid organism.
